web/server/system/shell.go

```diff
@@ -28,16 +28,17 @@ func NewShell(gobin, reportsPath string, coverage bool, defaultTimeout string) *
 	}
 }
 
-func (self *Shell) GoTest(directory, packageName string, arguments []string) (output string, err error) {
+func (self *Shell) GoTest(directory, packageName string, tags, arguments []string) (output string, err error) {
 	reportFilename := strings.Replace(packageName, "/", "-", -1)
 	reportPath := filepath.Join(self.reportsPath, reportFilename)
 	reportData := reportPath + ".txt"
 	reportHTML := reportPath + ".html"
+	tagsArg := "-tags=" + strings.Join(tags, ",")
```

Comment:
it's actually fine to have `-tags=` as a parameter to all of these tools. It's equivalent to the default (of not specifying -tags at all)